src/ai/ai-reranker.ts

```diff
@@ -50,19 +53,46 @@ export async function rerank({
   if (keywordResults.length === 0) {
     return vectorResults;
   }
-
-  const vectorRRFPromise = vectorResults.map(async (result, position) => {
-    return {
-      ...result,
-      score: await calculateRRF({ searchRank: position }),
-    };
+  //filtering low score value, 0.3 is suggested for vector search without HyDE logic
+  const filterdVectoResults = vectorResults.filter((vectorResult) => {
+    return vectorResult.score >= 0.3;
   });
-  const keywordRRFPromise = keywordResults.map(async (result, position) => {
+  const vectorRRFPromise = filterdVectoResults.map(async (result, position) => {
```

Comment:
replace your map with for loop

it's more efficient (I understand it goes against everything I said before, but I learned this while handling tons of files)

Additionally, it will reduce the number of loop done (each filter and map is a loop behind the scene)